src/libstore/uds-remote-store.cc

```diff
@@ -27,37 +26,25 @@ std::string UDSRemoteStoreConfig::doc()
 }
 
 
-UDSRemoteStore::UDSRemoteStore(const Params & params)
+UDSRemoteStore::UDSRemoteStore(std::string_view scheme, std::string_view authority, const Params & params)
     : StoreConfig(params)
     , LocalFSStoreConfig(params)
     , RemoteStoreConfig(params)
-    , UDSRemoteStoreConfig(params)
+    , UDSRemoteStoreConfig(scheme, authority, params)
     , Store(params)
     , LocalFSStore(params)
     , RemoteStore(params)
 {
 }
 
-
-UDSRemoteStore::UDSRemoteStore(
-    std::string_view scheme,
-    PathView socket_path,
-    const Params & params)
-    : UDSRemoteStore(params)
+std::string UDSRemoteStore::getPathOrDefault() const
 {
-    if (!socket_path.empty())
-        path.emplace(socket_path);
+    return path.value_or(settings.nixDaemonSocketFile);
 }
 
-
 std::string UDSRemoteStore::getUri()
 {
-    if (path) {
-        return std::string("unix://") + *path;
-    } else {
-        // unix:// with no path also works. Change what we return?
-        return "daemon";
-    }
+    return std::format("{}://{}", scheme, getPathOrDefault());
```

Comment:
We don't have this on macOS yet, I'm afraid. better just use `+`?

Reply:
I am not sure whether we want to stop returning `"daemon"` when the path is not the default, either.